skbio/stats/composition.py

```diff
@@ -608,10 +611,329 @@ def centralize(mat):
+    of zero cannot be computed.  While this is an unsolved problem, many
+    studies have shown promising results by replacing the zeros with pseudo
+    counts. This can be also be done via the ``multiplicative_replacement``
+    method.
+
+
+    References
+    ----------
+    .. [1] Mandal et al. "Analysis of composition of microbiomes: a novel
+       method for studying microbial composition", Microbial Ecology in Health
+       & Disease, (2015), 26.
+    .. [2] Holm, S. "A simple sequentially rejective multiple test procedure".
+       Scandinavian Journal of Statistics (1979), 6.
+
+    Examples
+    --------
+    First import all of the necessary modules:
+
+    >>> from skbio.stats.composition import ancom
+    >>> import pandas as pd
+
+    Now let's load in a pd.DataFrame with 6 samples and 7 unknown bacteria:
+
+    >>> table = pd.DataFrame([[12, 11, 10, 10, 10, 10, 10],
+    ...                       [9,  11, 12, 10, 10, 10, 10],
+    ...                       [1,  11, 10, 11, 10, 5,  9],
+    ...                       [22, 21, 9,  10, 10, 10, 10],
+    ...                       [20, 22, 10, 10, 13, 10, 10],
+    ...                       [23, 21, 14, 10, 10, 10, 10]],
+    ...                      index=['s1','s2','s3','s4','s5','s6'],
+    ...                      columns=['b1','b2','b3','b4','b5','b6','b7'])
+
+    Then create a grouping vector.  In this scenario, there
+    are only two classes, and suppose these classes correspond the
+    treatment due to a drug and a control.  The first three samples
+    are controls and the last three samples are treatments.
+
+    >>> grouping = pd.Series([0, 0, 0, 1, 1, 1],
+    ...                      index=['s1','s2','s3','s4','s5','s6'])
+
+    Now run `ancom` and see if there are any features that have any
+    significant differences between the treatment and the control.
+
+    >>> results = ancom(table, grouping)
+    >>> results['W']
+    b1    0
+    b2    4
+    b3    1
+    b4    1
+    b5    1
+    b6    0
+    b7    1
+    Name: W, dtype: int64
+
+    The W-statistic is the number of features that a single feature is tested
+    to be significantly different against.  In this scenario, b2 was detected
+    to have significantly different abundances compared to four of the other
+    species. To summarize the results from the W-statistic, let's take a look
+    at the results from the hypothesis test:
+
+    >>> results['reject']
+    b1    False
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    Name: reject, dtype: bool
+
+    From this we can conclude that only b2 was significantly
+    different between the treatment and the control.
+
+    """
+    if set(grouping.index) != set(table.index):
+        raise ValueError('There are row names not consistent between'
+                         '`grouping` and `table`')
+
+    # Check for negative values
+    closure(table)
```

Comment:
Previous code was reassigning so that closure operation happened. Is that not necessary anymore?

If all you need to do is ensure values are greater than zero, use:

```python
if np.any(table <= 0):
    raise ValueError(...)
```

And then remove check for zeros below, no need to take multiple passes through `table`.